scripts/import_transformer.py

```diff
@@ -0,0 +1,417 @@
+
+    inp_seq_name = "{}_input".format(ENCODER_NAME)
+    inp_seq = EmbeddedSequence(
+        name=inp_seq_name,
+        vocabulary=vocabulary,
+        data_id="source_wp",
+        embedding_size=hparams["embedding_size"])
+
+    encoder = TransformerEncoder(
+        name=ENCODER_NAME,
+        input_sequence=inp_seq,
+        ff_hidden_size=hparams["ff_hidden_size"],
+        depth=hparams["depth"],
+        n_heads=hparams["n_heads"])
+
+    encoder_ini = ENCODER_TEMPLATE.format(
+        inp_seq_name, hparams["embedding_size"], hparams["max_length"],
+        ENCODER_NAME, hparams["ff_hidden_size"], hparams["depth"],
+        hparams["n_heads"])
+
+    return encoder, vocabulary, "\n".join([vocabulary_ini, encoder_ini])
+
+
+DECODER_TEMPLATE = """\
+[decoder]
+class=decoder.transformer.TransformerDecoder
+name="{}"
+vocabulary=<vocabulary>
+data_id="target"
+encoder=<encoder>
+ff_hidden_size={}
+n_heads_self={}
+n_heads_enc={}
+depth={}
+embedding_size={}
+embdeddings_source=<input_sequence>
+max_output_len=50
+dropout_keep_prob=1.0
+attention_dropout_keep_prob=1.0
+"""
+
+
+def build_decoder(hparams: Dict,
+                  encoder: TransformerEncoder,
+                  vocab: Vocabulary) -> Tuple[
+        TransformerDecoder, str]:
+    decoder = TransformerDecoder(
+        name=DECODER_NAME,
+        encoder=encoder,
+        vocabulary=vocab,
+        data_id="target_wp",
+        ff_hidden_size=hparams["ff_hidden_size"],
+        n_heads_self=hparams["n_heads"],
+        n_heads_enc=hparams["n_heads"],
+        depth=hparams["depth"],
+        embeddings_source=encoder.input_sequence,
+        embedding_size=hparams["embedding_size"],
+        max_output_len=50)
+
+    decoder_ini = DECODER_TEMPLATE.format(
+        DECODER_NAME, hparams["ff_hidden_size"], hparams["n_heads"],
+        hparams["n_heads"], hparams["depth"], hparams["embedding_size"])
+
+    return decoder, "\n".join([decoder_ini])
+
+
+def build_model(hparams: Dict,
+                vocab_path: str) -> Tuple[
+        TransformerEncoder, TransformerDecoder, str]:
+    encoder, vocab, encoder_cfg = build_encoder(hparams, vocab_path)
+    decocer, decoder_cfg = build_decoder(hparams, encoder, vocab)
+
+    ini = "\n".join([encoder_cfg, decoder_cfg])
+
+    return ini
+
+
+def load_hparams(path: str) -> Dict:
+    with open(path, "r", encoding="utf-8") as f_json:
+        contents = json.load(f_json)
+
+    hparams = {
+        "n_heads": contents["num_heads"],
+        "ff_hidden_size": contents["filter_size"],
+        "embedding_size": contents["hidden_size"],
+        "max_length": contents["max_length"],
+        "label_smoothing": contents["label_smoothing"],
+        "depth": contents["num_hidden_layers"]
+    }
+
+    # TODO: check, whether the hparams that we do not set in NeuralMonkey
+    # are set to correct values
+
+    return hparams
+
+def assign_vars(hparams: Dict, np_vars: Dict) -> List[tf.Tensor]:
```

Comment:
Tohle bych přejmenoval na něco jako `get_asign_ops`, aby bylo jasné, že to nic samo neassignuje, jneom dělá ty opy. A doctring :-)